cuda.spec

```diff
@@ -14,78 +13,76 @@ AutoReq: no
 mkdir -p %_builddir/tmp
 /bin/sh %{SOURCE0} --silent --tmpdir %_builddir/tmp --extract %_builddir
 # extracts:
-# %_builddir/NVIDIA-Linux-x86_64-396.37.run
-# %_builddir/cuda-linux.9.2.148-24330188.run
-# %_builddir/cuda-samples.9.2.148-24330188-linux.run
+# %_builddir/NVIDIA-Linux-x86_64-410.48.run
+# %_builddir/cuda-linux.10.0.130-24817639.run
+# %_builddir/cuda-samples.10.0.130-24817639-linux.run
 
 # extract and repackage the CUDA runtime, tools and stubs
 /bin/sh %_builddir/%{n}-linux.%{realversion}-*.run -noprompt -nosymlink -tmpdir %_builddir/tmp -prefix %_builddir
 
-# Patch 1 (Released Aug 6, 2018)
-# CUDA 9.2 Patch Update: This update includes performance improvements to cuBLAS GEMM APIs and bug fixes for CUPTI and cuda-gdb.
-# See the CUDA 9.2 release notes for more details.
-/bin/sh %{SOURCE1} --silent --accept-eula --tmpdir %_builddir/tmp --installdir %_builddir
-rm -rf %_builddir/lib64/libcublas.so.9.2.148
-rm -rf %_builddir/lib64/libnvblas.so.9.2.148
-rm -rf %_builddir/lib64/libcuinj64.so.9.2.148
-rm -rf %_builddir/extras/CUPTI/lib64/libcupti.so.9.2.148
-
 mkdir -p %{i}/lib64
-# package only runtime and device static libraries
-cp -ar %_builddir/lib64/libcudart_static.a %{i}/lib64/
-cp -ar %_builddir/lib64/libcudadevrt.a %{i}/lib64/
-cp -ar %_builddir/lib64/lib*_device.a %{i}/lib64/
-rm -rf %_builddir/lib64/lib*.a
-
-# do not package dynamic libraries for which we have stubs
-rm -rf %_builddir/lib64/libcublas.so*
-rm -rf %_builddir/lib64/libcufft.so*
-rm -rf %_builddir/lib64/libcufftw.so*
-rm -rf %_builddir/lib64/libcurand.so*
-rm -rf %_builddir/lib64/libcusolver.so*
-rm -rf %_builddir/lib64/libcusparse.so*
-rm -rf %_builddir/lib64/libnpp*.so*
-rm -rf %_builddir/lib64/libnvgraph.so*
-rm -rf %_builddir/lib64/libnvrtc.so*
-rm -rf %_builddir/lib64/libnvrtc-builtins.so*
+mkdir -p %{i}/bin
+
+# package only the runtime static libraries
+mv %_builddir/lib64/libcudart_static.a %{i}/lib64/
```

Comment:
@smuzaffar moving the files instead of copying them should make the build faster and use less disk space.
Can you think of any reasons why we would not want to do it ?

Reply:
+1
No, there should not be any reason to not do that.
While you are at it, could you remove, create and use %_builddir/build to extract the files. Removing and creating it will make sure that in case of re-build (due to previous failures) we hae a clean build directory.

Reply:
Do you mean `%_builddir/build` instead of `%_builddir` ?

Reply:
Also, should I move the repackaging from `%install` to `%build` ?

Reply:
yes `%_builddir/build` instead of `%_builddir`. Extracting directly in %_builddir might override some files which cmsBuild generates e.g log file.

No, repacking in `%install` is fine.